server.go

```diff
@@ -130,6 +134,14 @@ func (server *Server) Stop() {
 		server.listener.Wait()
 	}
 
+	server.connections.Range(func(key, _ interface{}) bool {
+		c := key.(*Connection)
+		c.closeMutex.Lock()
+		defer c.closeMutex.Unlock()
+		c.closeInternal()
```

Comment:
why not just `c.Close()`?

Reply:
Close() puts connection back in the pool. Makes sense in normal life, but no sense when it is stopping.